server/src-lib/Hasura/Server/App.hs

```diff
@@ -65,9 +66,11 @@ import qualified Hasura.Server.PGDump                   as PGD
 data SchemaCacheRef
   = SchemaCacheRef
   { _scrLock     :: MVar ()
+  -- ^ TODO consider removing in place of `_scrCache :: MVar ...` if it's okay
+  -- or in fact correct to block during schema update in e.g. _wseGCtxMap.
```

Comment:
The idea behind explicit locking here is to 
1. Allow maximum throughput for serving requests (/v1/graphql) (as each request reads the current schemacache) 
2. We don't want to process more than one request at any point of time which would modify the schema cache as such queries are expensive.

Reply:
> if it's okay or in fact correct to block during schema update in e.g. _wseGCtxMap

It is theoretically possible to have a situation (in between [building new schemacache](https://github.com/hasura/graphql-engine/pull/3860/files#diff-faa5431efd1e2b8a934e9a860b158b3dR131) and [before writing it to the IORef](https://github.com/hasura/graphql-engine/pull/3860/files#diff-faa5431efd1e2b8a934e9a860b158b3dR132)) where we serve a request with a stale schemacache but I guess it is an okay trade-off to pay for a higher throughput (I remember doing a bunch of benchmarks to test this hypothesis).

Reply:
I'll add this additional information, thanks!